staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types.go

```diff
@@ -139,3 +140,84 @@ type CustomResourceDefinitionList struct {
 	// Items individual CustomResourceDefinitions
 	Items []CustomResourceDefinition
 }
+
+// CustomResourceValidation is a list of validation methods for CustomResources.
+type CustomResourceValidation struct {
+	// OpenAPIV3Schema is the OpenAPI v3 schema to be validated against.
+	OpenAPIV3Schema *JSONSchemaProps
```

Comment:
I think @lavalamp suggested we change this to a json string instead of representing Schema in our struct. Does that make things easier? (e.g. does problem of protobuf @smarterclayton pointed out will fade away if we do that) I think representing this in our API made lots of trouble for us as we don't really benefit from it. We can accept an string (or a free object) and deserialize it to JSONSchemaProps at our backend code. WDYT?